packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -441,6 +456,7 @@ export class TextFieldBase extends BaseComponent<ITextFieldProps, ITextFieldStat
         aria-invalid={!!this.state.errorMessage}
         aria-label={this.props.ariaLabel}
         readOnly={this.props.readOnly}
+        required={this.props.required}
```

Comment:
Also, the `required` native prop will work only if the TextField is placed within a `form` and clicking the submit button. Good to have it though... unless the `getNativeProps` is already setting it at top and you are just reassigning it here.